4.analyze_data/IoU.ipynb

```diff
@@ -0,0 +1,78 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Perform Intersection over Union (IoU) analysis on object masks "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# iou function from stackoverflow user gil.fernandes\n",
+    "# https://stackoverflow.com/questions/66595055/fastest-way-of-computing-binary-mask-iou-with-numpy\n",
+    "\n",
+    "import torch\n",
+    "import cv2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "tensor(1.)\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Intersection over Union function (IoU) utilizing ground truth (gt) and test binary masks \n",
+    "def iou(gt, test):\n",
+    "    intersection = (gt * test).sum()\n",
+    "    if intersection == 0:\n",
+    "        return 0.0\n",
+    "    union = torch.logical_or(gt, test).to(torch.int).sum()\n",
+    "    return intersection / union \n",
+    "\n",
+    "# Read binary masks \n",
+    "mask1 = cv2.imread('/home/maggiekeating/M1_B1_01_1_2_DAPI_001_nuclei_mask.tiff', 0)\n",
+    "mask2 = cv2.imread('/home/maggiekeating/M2_B1_01_1_2_DAPI_001_nuclei_mask.tiff', 0)\n",
+    "\n",
+    "# Convert binary masks from numpy ndarray to pytorch tensor \n",
+    "mask1_conv = torch.from_numpy(mask1)\n",
+    "mask2_conv = torch.from_numpy(mask2)\n",
+    "\n",
+    "print(iou(mask1_conv, mask2_conv))"
```

Comment:
might be on your to-do list but consider outputting these iou stats for downstream analysis

Reply:
I was thinking about outputting them to a CSV. Is that a good way to go about it?